aeon/dj_pipeline/utils/plotting.py

```diff
@@ -197,3 +199,144 @@ def plot_average_time_distribution(session_keys):
+    color = "food_patch_description" if per_food_patch else None
+
+    if per_food_patch:  # split by food patch
+        visit_per_day_df = (
+            (
+                (VisitSummary.FoodPatch & visit_key)
+                * acquisition.ExperimentFoodPatch.proj("food_patch_description")
+            )
+            .fetch(format="frame")
+            .reset_index()
+        )
+    else:
+        visit_per_day_df = (
+            ((VisitSummary & visit_key)).fetch(format="frame").reset_index()
+        )
+        if not attr.startswith("total"):
+            attr = "total_" + attr
+
+    visit_per_day_df["subject"] = "_".join([subject, visit_start.strftime("%m%d")])
+    visit_per_day_df["day"] = (
+        visit_per_day_df["visit_date"] - visit_per_day_df["visit_date"].min()
+    )
+    visit_per_day_df["day"] = visit_per_day_df["day"].dt.days
+
+    fig = px.line(
+        visit_per_day_df,
+        x="day",
+        y=attr,
+        color=color,
+        markers=True,
+        labels={attr: attr.replace("_", " ")},
+        hover_name="visit_date",
+        hover_data=["visit_date"],
+        width=700,
+        height=400,
+        template="simple_white",
+        title=visit_per_day_df["subject"][0],
+    )
+    fig.update_traces(mode="markers+lines", hovertemplate=None)
+    fig.update_layout(
+        legend_title="", hovermode="x", yaxis_tickformat="digits", yaxis_range=[0, None]
+    )
+
+    return fig
+
+
+def plot_foraging_bouts(
+    visit_key,
+    wheel_dist_crit=None,
+    min_bout_duration=None,
+    using_aeon_io=False,
+):
+    """plot the number of foraging bouts per visit
+
+    Args:
+        visit_key (dict) : Key from the VisitTimeDistribution table
+        wheel_dist_crit (int) : Minimum wheel distance travelled (in cm)
+        min_bout_duration (int) : Minimum foraging bout duration (in seconds)
+        using_aeon_io (bool) : Use aeon api to calculate wheel distance. Otherwise use datajoint tables. Defaults to False.
+
+    Returns:
+        fig: Figure object
+
+    Examples:
+        >>> fig = plot_foraging_bouts(visit_key, wheel_dist_crit=1, min_bout_duration=1)
+    """
+
+    subject, visit_start = (
+        visit_key["subject"],
+        visit_key["visit_start"],
+    )
+
+    visit_per_day_df = (
+        (
+            (VisitTimeDistribution.FoodPatch & visit_key)
+            * acquisition.ExperimentFoodPatch.proj("food_patch_description")
+        )
+        .fetch(format="frame")
+        .reset_index()
+    )
+
+    visit_per_day_df["subject"] = "_".join([subject, visit_start.strftime("%m%d")])
+    visit_per_day_df["day"] = (
+        visit_per_day_df["visit_date"] - visit_per_day_df["visit_date"].min()
+    )
+    visit_per_day_df["day"] = visit_per_day_df["day"].dt.days
+    visit_per_day_df["foraging_bouts"] = visit_per_day_df.apply(
+        VisitTimeDistribution._get_foraging_bouts,
+        args=(wheel_dist_crit, min_bout_duration, using_aeon_io),
+        axis=1,
+    )
```

Comment:
now I see how `VisitTimeDistribution._get_foraging_bouts()` is used. If this is the case, then perhaps no need to make this a class method, just let this be a private function here in this `plotting` module, and call it here.
Because in order to use it, ones would need to have a very specifically made pandas dataframe as input.